CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,96 @@
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+#include <QUdpSocket>
+#include "InternetConnectionService.h"
+#include <QDebug>
+#include "../../InfrastructureLayer/Settings/I_Settings.h"
+#include <QTextStream>
+#include <QTimer>
+namespace
+{
+    quint32 TIMEOUT_MILLISECONDS = 120000;
+    quint32 SLEEP_TIME_MILLISECONDS = 2000;
+    quint32 numberOfCalls = 0;
+}
+
+
+InternetConnectionService::InternetConnectionService(
+    QString exchangeName
+    ,  QString ipAddress
+    ,  quint16 udpPort)
+    : exchangeName_(exchangeName)
+    , ipAddress_(ipAddress)
+    , udpPort_(udpPort)
+{
+    QObject::connect(this, SIGNAL(setupChannelSignal()), this, SLOT(setupChannel()));
+
+    qtimer_ = new QTimer(this);
+    qtimer_->setSingleShot(true);
+    connect(qtimer_, SIGNAL(timeout()), this, SLOT(setupChannel()));
+    connectToDataSource();
+}
+
+InternetConnectionService::~InternetConnectionService()
+{
+}
+
+
+void InternetConnectionService::setupChannel()
+{
+    qWarning() << "UdpMessageForwarder: Attempting to connect" << ipAddress_ << udpPort_;
+
+    try
+    {
+        channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), udpPort_);
+    }
+    catch (AmqpClient::AmqpException::exception&)
+    {
+        if (channel_ == NULL)
+        {
+            if (numberOfCalls == (TIMEOUT_MILLISECONDS / SLEEP_TIME_MILLISECONDS))
+            {
+                qWarning() << "UdpMessageForwarder timed out waiting for connection to broker";
+                throw;
+            }
+        }
+        else
+        {
+            qWarning() << "bad alloc";
```

Comment:
Imagine if the program crashes and this is all is given. You're gonna want a more descriptive error.